Formula/glib.rb

```diff
@@ -18,28 +19,52 @@ class Glib < Formula
 
   depends_on "meson" => :build
   depends_on "ninja" => :build
+  depends_on "pcre" => :build
   depends_on "pkg-config" => :build
-  depends_on "gettext"
-  depends_on "libffi"
-  depends_on "pcre"
+  depends_on "libglib"
   depends_on "python@3.9"
 
-  on_linux do
-    depends_on "util-linux"
-  end
-
   # replace several hardcoded paths with homebrew counterparts
   patch do
     url "https://raw.githubusercontent.com/Homebrew/formula-patches/43467fd8dfc0e8954892ecc08fab131242dca025/glib/hardcoded-paths.diff"
     sha256 "d81c9e8296ec5b53b4ead6917f174b06026eeb0c671dfffc4965b2271fb6a82c"
   end
 
+  def script
+    <<~EOS
+      import pickle as pkl
+      from mesonbuild.minstall import load_install_data
+      installdata = load_install_data('meson-private/install.dat')
+      for attrname in ('data', 'emptydir', 'headers', 'install_scripts', 'install_subdirs', 'man', 'symlinks', 'targets'):
+          attr = getattr(installdata, attrname)
+          attr = list(filter(lambda data: all('opt/libglib' not in dataattr
+                                              for _, dataattr in vars(data)
+                                              if isinstance(dataattr, str)),
+                            attr))
+          setattr(installdata, attrname, attr)
+      with open('meson-private/install.dat', mode='wb') as file:
+          pkl.dump(file, installdata)
+    EOS
+  end
```

Comment:
This approach would be really hacky that modifies a binary file created by `meson` (create by `pickle.dump` from Python). I have read the source code of `meson`, I don't find any other way to do this.

The file targets can be ignored by changing the `mtime` of files (make it older than those files in `libglib`). But there are also some symlinks in which we cannot do the same things. Then it will raise `PermissionError` due to sandboxing.

The only way to not write things to `libglib` is overwriting the install data of `meson`.